File: app/src/main/java/com/lazysoul/kotlinwithandroid/ui/detail/DetailMvpPresenter.java
package com.lazysoul.kotlinwithandroid.ui.detail;

import com.lazysoul.kotlinwithandroid.common.BaseMvpPresenter;
import com.lazysoul.kotlinwithandroid.common.BaseMvpView;

import android.content.Intent;

/**
 * Created by Lazysoul on 2017. 7. 15..
 */

interface DetailMvpPresenter<MvpView extends BaseMvpView> extends BaseMvpPresenter<MvpView> {

    void loadTodo(Intent intent);

    void onTextChanged(String s);

    boolean isFixed();

    void saveTodo(String text);
}

File: app/src/main/kotlin/com/lazysoul/kotlinwithandroid/ui/detail/DetailMvpPresenter.kt
package com.lazysoul.kotlinwithandroid.ui.detail

import android.content.Intent
import com.lazysoul.kotlinwithandroid.common.BaseMvpPresenter
import com.lazysoul.kotlinwithandroid.common.BaseMvpView

/**
 * Created by Lazysoul on 2017. 7. 15..
 */

interface DetailMvpPresenter<in MvpView : BaseMvpView> : BaseMvpPresenter<MvpView> {

    val isFixed: Boolean

    fun loadTodo(intent: Intent)

    fun onTextChanged(s: String)

    fun saveTodo(text: String)
}
